tests/queries/0_stateless/01386_negative_float_constant_key_condition.sql

```diff
@@ -15,7 +15,7 @@ INSERT INTO t0 VALUES (0, 0);
 
 SELECT t0.c1 FROM t0 WHERE NOT (t0.c1 OR (t0.c0 AND -1524532316));
 SELECT t0.c1 FROM t0 WHERE NOT (t0.c1 OR (t0.c0 AND -1.0)); -- { serverError 70 }
-SELECT t0.c1 FROM t0 WHERE NOT (t0.c1 OR (t0.c0 AND inf));
-SELECT t0.c1 FROM t0 WHERE NOT (t0.c1 OR (t0.c0 AND nan));
+SELECT t0.c1 FROM t0 WHERE NOT (t0.c1 OR (t0.c0 AND inf)); -- { serverError ILLEGAL_TYPE_OF_ARGUMENT }
```

Comment:
@alexey-milovidov It seems you've added `inf` and `nan` here intentionally some time ago. Could you clarify the motivation behind it? It looks at least controversial

Reply:
The motivation is the following:
1. Not only UInt8 is supported in the arguments of logical operators, but other numeric types as well.
2. It is reasonable for all numeric types which include 0 and 1.
3. The behavior should be the same as in C/C++: when the value is non-zero, it behaves as true.
4. That explains why it should work for Floating types as well.
5. The behavior for `inf` and `nan` does not have to be as well defined, but we should have at least one test to assert it.
6. That's why the text exists.

Reply:
7. I don't mind if you change it, and add a note to the changelog.